src/com/dojo/foundations/parking/TheGarage.java

```diff
@@ -89,81 +49,71 @@ public static void main(String[] args) throws Exception {
                     System.out.println("Write the motorcycle plate:");
                     String motorcyclePlate = scanner.readLine();
                     Matcher matcherMotorcycle = PATTERN_MOTORCYCLE_PLATE.matcher(motorcyclePlate);
-                    if (!matcherMotorcycle.find()) {
-                        System.out.println("The plate is invalid");
-                        return;
-                    }
+                    evaluatePlate(matcherMotorcycle);
```

Comment:
I think this method will not escape the execution of the loop when the plate is not correct.

According to the regular expressions, the plate structures are:

Motorcycle ABC12T
Car ABC123
Truck T1234

These are the plate formats in Colombia.